database/merkledb/tree.go

```diff
@@ -0,0 +1,281 @@
+package merkledb
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/database/versiondb"
+
+	"github.com/ava-labs/avalanchego/database/leveldb"
+
+	"github.com/ava-labs/avalanchego/database/memdb"
+
+	"github.com/ava-labs/avalanchego/database"
+)
+
+// Tree holds the tree data
+type Tree struct {
+	closed      bool
+	persistence *Persistence
+}
+
+// Has returns whether the key exists in the tree
+func (t *Tree) Has(key []byte) (bool, error) {
+	if t.isClosed() != nil {
+		return false, t.isClosed()
+	}
+
+	rootNode, err := t.persistence.GetRootNode()
```

Comment:
In general, yes. I think I had it like that before the Forest Trie.
Using the persistence abstraction helps to separte the behavior of the RootNode, such as picking it from the repo from the behaviour of the Tree.

I removed it because having a link in the tree meant maintaing the awereness - at the tree leve - of which root node we're doing operations on.